addons/binding/org.openhab.binding.sensebox/src/main/java/org/openhab/binding/sensebox/handler/SenseBoxHandler.java

```diff
@@ -0,0 +1,373 @@
+import org.openhab.binding.sensebox.model.SenseBoxLocation;
+import org.openhab.binding.sensebox.model.SenseBoxSensor;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.google.gson.Gson;
+
+/**
+ * The {@link SenseBoxHandler} is responsible for handling commands, which are
+ * sent to one of the channels.
+ *
+ * @author Hakan Tandogan - Initial contribution
+ */
+
+public class SenseBoxHandler extends BaseThingHandler {
+    private Logger logger = LoggerFactory.getLogger(SenseBoxHandler.class);
+
+    protected SenseBoxConfiguration thingConfiguration;
+
+    private Gson gson = new Gson();
+
+    private SenseBoxData data = new SenseBoxData();
+
+    ScheduledFuture<?> refreshJob;
+
+    private final int CACHE_EXPIRY = 10 * 1000; // 10s
+
+    public SenseBoxHandler(Thing thing) {
+        super(thing);
+    }
+
+    @Override
+    public void initialize() {
+        logger.debug("Start initializing!");
+
+        thingConfiguration = getConfigAs(SenseBoxConfiguration.class);
+        String thingUid = getThing().getUID().toString();
+        thingConfiguration.setThingUid(thingUid);
+
+        logger.debug("Thing Configuration {} initialized {}", thingConfiguration.getThingUid(),
+                thingConfiguration.getSenseBoxId());
+
+        String offlineReason = "";
+        boolean validConfig = true;
+
+        if (StringUtils.trimToNull(thingConfiguration.getSenseBoxId()) == null) {
+            offlineReason = "senseBox ID is mandatory and must be configured";
+            logger.error("{}, disabling thing '{}'", offlineReason, thingUid);
+            validConfig = false;
+        }
+
+        if (thingConfiguration.getRefreshInterval() < MINIMUM_UPDATE_INTERVAL) {
+            logger.info("Refresh interval is much too small, setting to default of {} seconds",
+                    MINIMUM_UPDATE_INTERVAL);
+            thingConfiguration.setRefreshInterval(MINIMUM_UPDATE_INTERVAL);
+        }
+
+        if (validConfig) {
+            logger.debug("Thing Configuration: {}", thingConfiguration.toString());
+            updateStatus(ThingStatus.ONLINE);
+            startAutomaticRefresh();
+        } else {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR, offlineReason);
+        }
+
+        logger.debug("Thing {} initialized {}", getThing().getUID(), getThing().getStatus());
+    }
+
+    @Override
+    public void handleCommand(ChannelUID channelUID, Command command) {
+        if (ThingStatus.ONLINE == thing.getStatus()) {
+            if (command instanceof RefreshType) {
+                try {
+                    data = fetchData();
+                    publishDataForChannel(channelUID);
+                } catch (IOException e) {
+                    logger.warn("Exception occurred during fetching data: {}", e.getMessage(), e);
+                } catch (Exception e) {
+                    logger.debug("Exception occurred during execution: {}", e.getMessage(), e);
```

Comment:
In which situations can this happen? Might you also want to set the ThingStatus to OFFLINE here?